src/Symfony/Component/JsonPath/JsonPathUtils.php

```diff
@@ -85,4 +85,82 @@ public static function findSmallestDeserializableStringAndPath(array $tokens, mi
             'tokens' => $remainingTokens,
         ];
     }
+
+    public static function unescapeString(string $str, string $quoteChar): string
+    {
+        if ('"' === $quoteChar) {
+            // try JSON decoding first for unicode sequences
+            $jsonStr = '"' . $str . '"';
+            $decoded = json_decode($jsonStr, true);
+
+            if (null !== $decoded) {
+                return $decoded;
+            }
+        }
+
+        $result = '';
+        $length = strlen($str);
+
+        for ($i = 0; $i < $length; $i++) {
+            if ('\\' === $str[$i] && $i + 1 < $length) {
+                $nextChar = $str[$i + 1];
+
+                $result .= match ($nextChar) {
+                    '"' => '"',
+                    "'" => "'",
+                    '\\' => '\\',
+                    '/' => '/',
+                    'b' => "\b",
+                    'f' => "\f",
+                    'n' => "\n",
+                    'r' => "\r",
+                    't' => "\t",
+                    'u' => self::unescapeUnicodeSequence($str, $length, $i),
+                    default => $str[$i], // keep the backslash
+                };
+
+                if ('u' !== $nextChar) {
+                    ++$i;
+                }
```

Comment:
the u case will need to increment one less
```suggestion
                ++$i;
```